plugins/catalog-backend-module-github/src/lib/github.ts

```diff
@@ -325,6 +334,66 @@ export async function getTeamMembers(
   return { members };
 }
 
+/**
+ *
+ * Gets the specified yaml files from a github repo and returns the yaml data so it can be used
+ *  in a provider to be turned into entities.
+ *
+ * @param client - An octokit graphql client
+ * @param org - The slug of the org to read
+ * @param files - List of file entries pulled from the config
+ */
+
+export async function getOrganizationEntities(
+  client: typeof graphql,
+  org: string,
+  files: Array<string>,
+): Promise<{ repositories: Array<Repository | RepositoryWithFiles> }> {
+  let queryFiles = ``;
+
+  for (const file of files) {
+    queryFiles += `
+    ${camelCase(file)}: object(expression: "HEAD:${file}") {
```

Comment:
Probably better to base the label here on the index instead tbh, feeling that would be more robust?

From an encapsulation point of view I think it would be good if this method also handled the unpacking of these files from the response, so that we don't have this hidden contract of where the caller of `getOrganizationEntities` is supposed to find the files